python_modules/dagster/dagster/core/definitions/event_metadata/__init__.py

```diff
@@ -581,17 +628,87 @@ def validate_table(context, df):
         check.inst_param(asset_key, "asset_key", AssetKey)
         return DagsterAssetMetadataEntryData(asset_key)
 
+    @staticmethod
+    def table(records: List[TableRecord], schema: TableSchema) -> "TableMetadataEntryData":
+        """Static constructor for a metadata value wrapping arbitrary tabular data as
+        :py:class:`TableMetadataEntryData`. Can be used as the value type for the `metadata`
+        parameter for supported events. For example:
+
+        .. code-block:: python
+
+            @op
+            def emit_metadata(context):
+                yield ExpectationResult(
+                    success=not has_errors,
+                    label="is_valid",
+                    metadata={
+                        "errors": EventMetadata.table(
+                            records=[
+                                TableRecord(code="invalid-data-type", row=2, col="name"}]
+                            ],
+                            schema=TableSchema(
+                                columns=[
+                                    TableColumn(name="code", type="string"),
+                                    TableColumn(name="row", type="int"),
+                                    TableColumn(name="col", type="string"),
+                                ]
+                            )
+                        ),
+                    },
+                )
 
+        Args:
+            records (List[TableRecord]): The data as a list of records (i.e. rows).
+            schema (TableSchema): A schema for the table.
+        """
+        return TableMetadataEntryData(records, schema)
+
+    @staticmethod
+    def table_schema(
+        schema: TableSchema,
```

Comment:
what about the label?

Reply:
You're looking at `EventMetadata.table_schema`, not `EventMetadataEntry.table_schema` (that's at the bottom of the file, you'll see it takes a label). The `EventMetadata.<type>` methods don't take labels, they just proxy to the `XEventMetadataEntry` constructors.

FWIW I find this API superfluous and confusing for exactly the reason you got tripped up here. I wrote about it here: https://dagster.slack.com/archives/C02R1L5B74N/p1642789896026200?thread_ts=1642701522.022100&cid=C02R1L5B74N

Reply:
You're right - disregard my comment.